rpcserver.go

```diff
@@ -512,6 +512,10 @@ func MainRPCServerPermissions() map[string][]bakery.Op {
 			Entity: "offchain",
 			Action: "write",
 		}},
+		lnrpc.RegisterRPCMiddlewareURI: {{
+			Entity: "macaroon",
+			Action: "write",
```

Comment:
One thought here is that the read only interception also needs write perms?

Reply:
Yeah, I thought about that as well. But because even the read-only interception can be very intrusive from a privacy standpoint, I'd be nervous if you'd be able to intercept all traffic with just the default read-only macaroon. That's why I decided to need a write permission to _register_ your middleware.

Reply:
Does read-only macaroon mean "blanket allow all readonly calls"? Perhaps it'd be better to have "benign" macaroon (go ahead and suggest a better name) that would be recommended instead of readonly for clarity. Although it's quite a footgun to keep readonly and add privacy-intrusive permission.

Reply:
> Does read-only macaroon mean "blanket allow all readonly calls"? 

Yeah it has read permissions on all [Entities](https://github.com/lightningnetwork/lnd/blob/27b416935e3fa4fe3c0232994fa2283dfcdee344/rpcserver.go#L84). 

> That's why I decided to need a write permission to register your middleware.

Makes sense, you're _writing_ middleware that'll read only.

Reply:
> Makes sense, you're *writing* middleware that'll read only.

Ah, yes, sounds good.